src/utilities/fastify/server.ts

```diff
@@ -40,14 +41,21 @@ export async function makeFastify(appPlugin: FastifyPluginAsync, customLogger?:
 
   await server.register(fastifyMongoose);
 
-  server.setErrorHandler(function (_error, _request, reply) {
-    reply.status(HTTP_STATUS_CODES.INTERNAL_ERROR).send("Internal server error");
-  })
+  const internalServerError = 'Internal server error';
+  server.setErrorHandler(async (error, _request, reply) => {
+    if (error.statusCode !== undefined && error.statusCode < HTTP_STATUS_CODES.INTERNAL_ERROR) {
+      await reply.send(error);
```

Comment:
Since we're overriding the default error handler, we should also log an info log. See: https://www.fastify.io/docs/latest/Reference/Server/#seterrorhandler